tests/node-sync/run_full.go

```diff
@@ -51,56 +51,38 @@ func RunFullNode(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 		return err
 	}
 
-	bridgeTotalCh := make(chan int)
-	sub, err := syncclient.Subscribe(ctx, testkit.BridgeTotalTopic, bridgeTotalCh)
-	if err != nil {
-		return err
-	}
-
-	var bridgeTotal int
-	select {
-	case err = <-sub.Done():
-		if err != nil {
-			return err
-		}
-	case bridgeTotal = <-bridgeTotalCh:
-		err = <-syncclient.MustBarrier(ctx, testkit.BridgeStartedState, bridgeTotal).C
-		if err != nil {
-			return err
-		}
-	}
-
-	bridgeCh := make(chan *testkit.BridgeNodeInfo)
-	sub, err = syncclient.Subscribe(ctx, testkit.BridgeNodeTopic, bridgeCh)
+	bridgeCh := make(chan *testkit.BridgeNodeInfo, runenv.IntParam("bridge"))
+	sub, err := syncclient.Subscribe(ctx, testkit.BridgeNodeTopic, bridgeCh)
 	if err != nil {
 		return err
 	}
 
 	bridgeNode, err := func(total int) (*testkit.BridgeNodeInfo, error) {
-		for i := 0; i < total; i++ {
+		for {
```

Comment:
This anon func seems to be used over and over so can be extracted.
Together with Subsription btw to something like WaitForBridge(s)

Reply:
+1

Reply:
Moved this to run_bridge.go